### ISSUE TITLE
Showing only certain landmarks on android FaceLandmarker App

### ISSUE THREAD

**Issue by connormeaton** [googlesamples/mediapipe#372]
Hello, I am playing around with your Face Landmarker app for android.

I am very inexperienced with Koltin, so my apologies. I would like to display only certain landmarks on the face when running the app and I'm not sure where to start. For example, if I only wanted to show the left eye landmarks on the app, how would I do that?

In python, I would find the variable storing the facelandmark coordinates, and I would index it by these values to get the left eyelandmarks: [85, 246, 161, 160, 159, 158, 157, 173, 155, 154, 153, 145, 144, 163]. How would I do that in this koltin project? What variable contains the landmarks and how would I index it like this?

It looks like OverlayView.kt setResults function might be the place to start, but I'm really not sure. Any advise would be helpful. Thank you!

https://github.com/googlesamples/mediapipe/tree/main/examples/face_landmarker/android

**Comment by PaulTR** [comment]
Yup the OverlayView setResults function is where the landmarks to display are set! :) You can add a [filter](https://kotlinlang.org/docs/collection-filtering.html) before sending those results to limit it to only the landmarks that you want.

**Comment by connormeaton** [comment]
Thank you for your reply. That is really helpful. Can you critique this specific section? My approach is to modify setResults() to:
- create index of the only landmarks I want to keep
- filter results by these indices
- rewrite results with the landmark data for only the previously specified indices

`    fun setResults(
        faceLandmarkerResults: FaceLandmarkerResult,
        imageHeight: Int,
        imageWidth: Int,
        runningMode: RunningMode = RunningMode.IMAGE
    ) {
        results = faceLandmarkerResults
        val indexList = listOf(85, 246, 161, 160, 159, 158, 157, 173, 155, 154, 153, 145, 144, 163)
        val filteredNumbers = results.filterIndexed { index, value ->
            index in indexList
        }
        results = filteredNumbers
        this.imageHeight = imageHeight
        this.imageWidth = imageWidth

        scaleFactor = when (runningMode) {
            RunningMode.IMAGE,
            RunningMode.VIDEO -> {
                min(width * 1f / imageWidth, height * 1f / imageHeight)
            }
            RunningMode.LIVE_STREAM -> {
                // PreviewView is in FILL_START mode. So we need to scale up the
                // landmarks to match with the size that the captured images will be
                // displayed.
                max(width * 1f / imageWidth, height * 1f / imageHeight)
            }
        }
        invalidate()
    }`

However, I keep getting the error that the code cannot infer the type for this parameter (point to index and value). How should can I properly type these parameters, or, is there another way I should do this? 

Thank you!


The full error is below.

<html>Unresolved reference. None of the following candidates is applicable because of receiver type mismatch:<br/>public inline fun &lt;T&gt; Array&lt;out TypeVariable(T)&gt;.filterIndexed(predicate: (index: Int, TypeVariable(T)) -&gt; Boolean): List&lt;TypeVariable(T)&gt; defined in kotlin.collections<br/>public inline fun BooleanArray.filterIndexed(predicate: (index: Int, Boolean) -&gt; Boolean): List&lt;Boolean&gt; defined in kotlin.collections<br/>public inline fun ByteArray.filterIndexed(predicate: (index: Int, Byte) -&gt; Boolean): List&lt;Byte&gt; defined in kotlin.collections<br/>public inline fun CharArray.filterIndexed(predicate: (index: Int, Char) -&gt; Boolean): List&lt;Char&gt; defined in kotlin.collections<br/>public inline fun CharSequence.filterIndexed(predicate: (index: Int, Char) -&gt; Boolean): CharSequence defined in kotlin.text<br/>public inline fun DoubleArray.filterIndexed(predicate: (index: Int, Double) -&gt; Boolean): List&lt;Double&gt; defined in kotlin.collections<br/>public inline fun FloatArray.filterIndexed(predicate: (index: Int, Float) -&gt; Boolean): List&lt;Float&gt; defined in kotlin.collections<br/>public inline fun IntArray.filterIndexed(predicate: (index: Int, Int) -&gt; Boolean): List&lt;Int&gt; defined in kotlin.collections<br/>public inline fun LongArray.filterIndexed(predicate: (index: Int, Long) -&gt; Boolean): List&lt;Long&gt; defined in kotlin.collections<br/>public inline fun ShortArray.filterIndexed(predicate: (index: Int, Short) -&gt; Boolean): List&lt;Short&gt; defined in kotlin.collections<br/>public inline fun String.filterIndexed(predicate: (index: Int, Char) -&gt; Boolean): String defined in kotlin.text<br/>public inline fun UByteArray.filterIndexed(predicate: (index: Int, UByte) -&gt; Boolean): List&lt;UByte&gt; defined in kotlin.collections<br/>public inline fun UIntArray.filterIndexed(predicate: (index: Int, UInt) -&gt; Boolean): List&lt;UInt&gt; defined in kotlin.collections<br/>public inline fun ULongArray.filterIndexed(predicate: (index: Int, ULong) -&gt; Boolean): List&lt;ULong&gt; defined in kotlin.collections<br/>public inline fun UShortArray.filterIndexed(predicate: (index: Int, UShort) -&gt; Boolean): List&lt;UShort&gt; defined in kotlin.collections<br/>public inline fun &lt;T&gt; Iterable&lt;TypeVariable(T)&gt;.filterIndexed(predicate: (index: Int, TypeVariable(T)) -&gt; Boolean): List&lt;TypeVariable(T)&gt; defined in kotlin.collections<br/>public fun &lt;T&gt; Sequence&lt;TypeVariable(T)&gt;.filterIndexed(predicate: (index: Int, TypeVariable(T)) -&gt; Boolean): Sequence&lt;TypeVariable(T)&gt; defined in kotlin.sequences
